src/platform/h563xx/platform.c

```diff
@@ -176,3 +187,46 @@ void SysTick_Handler(void)
     HAL_IncTick();
     HAL_SYSTICK_IRQHandler();
 }
+
+/**
+ * @brief Get the clock speed for a specific clock type
+ *
+ * This function retrieves the clock speed for the specified clock type.
+ * It can be used to determine the frequency of various system clocks such as
+ * the system clock, AHB clock, and APB clocks.
+ *
+ * @param clk_type The type of clock to query (SYS_CLK, AHB_CLK, APB1_CLK,
+ * APB2_CLK, APB3_CLK)
+ *
+ * @return The clock speed in Hz for the specified clock type, or 0 if invalid
+ *         type is provided
+ */
+static uint32_t get_clock_speed(CLKType clk_type)
+{
+    if (clk_type == SYS_CLK) {
+        return HAL_RCC_GetSysClockFreq();
+    } else if (clk_type == AHB_CLK) {
+        return HAL_RCC_GetHCLKFreq();
+    } else if (clk_type == APB1_CLK) {
+        return HAL_RCC_GetPCLK1Freq();
+    } else if (clk_type == APB2_CLK) {
+        return HAL_RCC_GetPCLK2Freq();
+    } else if (clk_type == APB3_CLK) {
+        return HAL_RCC_GetPCLK3Freq();
+    } else {
+        return 0; // Invalid clock type
+    }
+}
+
+uint32_t get_peripheral_clock_speed(PeripheralClock clock)
```

Comment:
Public functions should be prefixed with the module name in caps.
```suggestion
uint32_t PLATFORM_get_peripheral_clock_speed(PeripheralClock clock)
```

Reply:
Ok, I will keep that in mind